app/models/metric/purging.rb

```diff
@@ -85,9 +87,32 @@ def self.purge_hourly(older_than, window = nil, total_limit = nil, &block)
   end
 
   def self.purge_realtime(older_than, window = nil, total_limit = nil, &block)
-    purge_by_date(older_than, "realtime", window, total_limit, &block)
+    truncate_child_tables(older_than)
   end
 
+  def self.truncate_child_tables(older_than)
+    target_hours = determine_target_hours(older_than, Time.now.utc)
+    return if target_hours.blank?
+
+    target_hours.each do |hour|
+      Metric.with_table_for_time(hour) do
+        Metric.truncate
+      end
+    end
+  end
+
+  def self.determine_target_hours(older_than, end_date = Time.now)
```

Comment:
Seems like the default value for `end_date` should match how you are invoking it above with `Time.now.utc` instead of just `Time.now`.

Reply:
Possibly just omit the above argument, and default `end_date` to `Time.now.utc`?